lampod/src/jsonrpc/channels.rs

```diff
@@ -8,3 +12,18 @@ pub fn json_list_channels(ctx: &LampoDeamon, request: &json::Value) -> Result<js
     let resp = ctx.channel_manager().list_channel();
     Ok(json::to_value(resp)?)
 }
+
+pub fn json_close_channel(ctx: &LampoDeamon, request: &json::Value) -> Result<json::Value, Error> {
+    log::info!("call for `closechannel` with request {:?}", request);
+    let request: request::CloseChannel = json::from_value(request.clone())?;
+    let res = ctx.channel_manager().close_channel(request);
+    let resp = match res {
+        Ok(resp) => Ok(resp),
+        Err(err) => Err(Error::Rpc(RpcError {
+            code: -1,
+            message: format!("{err}"),
+            data: None,
+        })),
+    };
+    Ok(json::to_value(resp?)?)
```

Comment:
can you wait before returning the `LightningEvent::CloseChannelEvent`? see the funding channel RPC call for an example

Reply:
Okay we need to loop here right? Like we did in https://github.com/vincenzopalazzo/lampo.rs/pull/198#discussion_r1530730754